crates/pixi_api/src/context.rs

```diff
@@ -28,6 +36,64 @@ impl<I: Interface> WorkspaceContext<I> {
         crate::workspace::workspace::name::set(&self.interface, self.workspace.clone(), name).await
     }
 
+    pub async fn list_tasks(
+        &self,
+        environment: Option<EnvironmentName>,
+    ) -> miette::Result<HashMap<EnvironmentName, HashMap<TaskName, Task>>> {
+        crate::workspace::task::list_tasks(&self.interface, self.workspace.clone(), environment)
+            .await
+    }
+
+    pub async fn add_task(
+        &self,
+        name: TaskName,
+        task: Task,
+        feature: FeatureName,
+        platform: Option<Platform>,
+    ) -> miette::Result<()> {
+        crate::workspace::task::add_task(
+            &self.interface,
+            self.workspace.clone(),
+            name,
+            task,
+            feature,
+            platform,
+        )
+        .await
+    }
+
+    pub async fn alias_task(
+        &self,
+        name: TaskName,
+        task: Task,
+        platform: Option<Platform>,
+    ) -> miette::Result<()> {
+        crate::workspace::task::alias_task(
+            &self.interface,
+            self.workspace.clone(),
```

Comment:
Maybe let this (and other functions) take a reference instead? Or do you really *always* need an owned type?